core/src/main/scala/pureconfig/ProductHint.scala

```diff
@@ -0,0 +1,40 @@
+package pureconfig
+
+/**
+ * A trait that can be implemented to customize how case classes are read from and written to a config.
+ *
+ * @tparam T the type of case class for which this hint applies
+ */
+trait ProductHint[T] {
+
+  /**
+   * The mapping used to map field names to config keys
+   */
+  def fieldMapping: ConfigFieldMapping
```

Comment:
Changed it in c84305a3253abf7c04432038c6d8090923050d06. @melrief, let me know if you prefer the other way.